### PR TITLE
Fix: memory leak of the apiserver

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

There are two points:

1. The `json.Marshal` and `json.Unmarshal` consume much memory.
2. Not call the `Done` function of the queue. The app object will pile up in a queue.

Now:

image

image

|  Item  | Before | Now |
|  ----  | ----  | ----  |
| Application Number  | 700+ |  1.4K |
| Application Create/Update  | 3 request/second |   10 request/second |
| KubeVela Core Memory  | 600Mi+ | 1057Mi |
| APIServer Memory(Master) | 900Mi+ | 300Mi | 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/utils/pprof.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+	pprofServer := http.Server{
+		Addr:    pprofAddr,
+		Handler: mux,
+	}
+	defer func() {
+		ctx, cancelFunc := context.WithTimeout(ctx, 60*time.Second)
+		defer cancelFunc()
```
Comment (wonderflow) [inline pkg/utils/pprof.go:44]:
```suggestion
	go func() {
		<- ctx.Done()
```

Comment (wonderflow) [reply pkg/utils/pprof.go:44]: the outside context need to have a cancel func

File: cmd/core/main.go [hunk 2]
```diff
@@ -163,36 +162,7 @@ func main() {
...
-					klog.Error(err, "Failed to shutdown debug HTTP server")
-				}
-			}()
-
-			if err := pprofServer.ListenAndServe(); !errors.Is(http.ErrServerClosed, err) {
-				klog.Error(err, "Failed to start debug HTTP server")
-				panic(err)
-			}
-		}()
+		go pkgutils.EnablePprof(context.Background(), pprofAddr)
```
Comment (wonderflow) [inline cmd/core/main.go:165]: you must wrapper context.Background to have a cancel function

Comment (barnettZQG) [reply cmd/core/main.go:165]: I think there is no need to cancel the context. There are many `os.Exit(1)` function calls. There is no graceful exit control.

Comment (wonderflow) [reply cmd/core/main.go:165]: But you're actually writing dead code if you don't pass a useful context here.

image

Comment (barnettZQG) [reply cmd/core/main.go:165]: For the apiserver, There is a graceful exit control. So the common function provides this ability.

**Review by wonderflow — APPROVED** [review]

**Review by FogDong — APPROVED** [review]

**Pull request merged**